### PR TITLE
guix: Default to building for some supported HOSTs only

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
https://github.com/dashpay/dash/pull/6382#discussion_r1833266383

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: contrib/guix/guix-build [hunk 1]
```diff
@@ -73,8 +73,19 @@ mkdir -p "$VERSION_BASE"
...
+# arm64-apple-darwin
+# powerpc64-linux-gnu
+# powerpc64le-linux-gnu
+# riscv64-linux-gnu
+# x86_64-apple-darwin
+# x86_64-linux-gnu
+# x86_64-w64-mingw32
+
+# Default to building for some supported HOSTs only (overridable by environment)
+export HOSTS="${HOSTS:-x86_64-linux-gnu aarch64-linux-gnu riscv64-linux-gnu
```
Comment (knst) [inline contrib/guix/guix-build:88]: should we ask in community if anyone has powerpc64 or arm32 to test? maybe there's non-zero demand for it? So far as build succeed, probably the binaries may work as well

Comment (kwvg) [inline contrib/guix/guix-build:88]: Consider b58a3b6e3f78e9224eefab717bf0c6eb84374918

**Review by kwvg — CHANGES_REQUESTED** [review]

File: contrib/guix/guix-build (hunk 1 above)
Comment (UdjinM6) [reply contrib/guix/guix-build:88]: you mean in addition to changes in `contrib/guix/guix-build` or as a replacement?

Comment (kwvg) [reply contrib/guix/guix-build:88]: Replacement, I think we should keep upstream code intact as much as we can and add restrictions through CI or our containers

Comment (PastaPastaPasta) [reply contrib/guix/guix-build:88]: I disagree, mostly, because very little if any of my automations use the guis-start. They all use guix-buidl

Comment (kwvg) [reply contrib/guix/guix-build:88]: `./contrib/containers/guix/scripts/guix-start $(pwd)` should be a drop-in

Comment (UdjinM6) [reply contrib/guix/guix-build:88]: Needs a couple of additional fixes but, yeah, it could work. Pls check #6390.

**Comment by PastaPastaPasta** [comment]
I think I prefer this PRs approach. Sure it'll cause conflicts. But the conflicts will be very minimal / trivial to resolve. I prefer not needing to change my automation to run the guix-start instead.

**Comment by UdjinM6** [comment]
> I think I prefer this PRs approach. Sure it'll cause conflicts. But the conflicts will be very minimal / trivial to resolve. I prefer not needing to change my automation to run the guix-start instead.

Don't be lazy :P

Speaking seriously though, #6390 requires trivial changes in private build scripts - a removal of SDK download and a drop-in replacement for `guix-build`. This needs to be done only once and adds no technical debt to our public repo. It also aligns CI builds and our release process. I think I'm actually leaning more towards that solution now...

**Comment by PastaPastaPasta** [comment]
Reopen if desired?

**Pull request closed without merging**